src/storage/v3/shard_rsm.cpp

```diff
@@ -289,41 +289,30 @@ TypedValue ComputeExpression(DbAccessor &dba, const std::optional<memgraph::stor
   auto expr = ParseExpression(expression, storage);
 
   auto node_identifier = Identifier(std::string(node_name), false);
-  bool is_node_identifier_present = false;
   auto edge_identifier = Identifier(std::string(edge_name), false);
-  bool is_edge_identifier_present = false;
 
   std::vector<Identifier *> identifiers;
 
-  // TODO: use a visitor instead of string search
-  if (v_acc && expression.find(node_name) != std::string::npos) {
-    is_node_identifier_present = true;
-    identifiers.push_back(&node_identifier);
-  }
-  if (e_acc && expression.find(edge_name) != std::string::npos) {
-    is_edge_identifier_present = true;
-    identifiers.push_back(&edge_identifier);
-  }
-
   expr::SymbolGenerator symbol_generator(&symbol_table, identifiers);
   (std::any_cast<Expression *>(expr))->Accept(symbol_generator);
 
+  auto node_identifier_position_on_frame =
+      std::find_if(symbol_table.table().begin(), symbol_table.table().end(),
+                   [&node_identifier](const std::pair<int32_t, Symbol> &position_symbol_pair) {
+                     return position_symbol_pair.second.name() == node_identifier.name_;
+                   });
+  auto is_node_identifier_present = node_identifier_position_on_frame != symbol_table.table().end();
   if (is_node_identifier_present) {
-    // We look for the position of node_identifier in the frame.
-    auto position_on_frame = std::find_if(symbol_table.table().begin(), symbol_table.table().end(),
-                                          [&node_identifier](const std::pair<int32_t, Symbol> &position_symbol_pair) {
-                                            return position_symbol_pair.second.name() == node_identifier.name_;
-                                          });
-    MG_ASSERT(position_on_frame != symbol_table.table().end());
     frame[symbol_table.at(node_identifier)] = *v_acc;
   }
+
+  auto edge_identifier_position_on_frame =
+      std::find_if(symbol_table.table().begin(), symbol_table.table().end(),
+                   [&edge_identifier](const std::pair<int32_t, Symbol> &position_symbol_pair) {
+                     return position_symbol_pair.second.name() == edge_identifier.name_;
+                   });
+  auto is_edge_identifier_present = edge_identifier_position_on_frame != symbol_table.table().end();
   if (is_edge_identifier_present) {
-    // We look for the position of node_identifier in the frame.
-    auto position_on_frame = std::find_if(symbol_table.table().begin(), symbol_table.table().end(),
-                                          [&edge_identifier](const std::pair<int32_t, Symbol> &position_symbol_pair) {
-                                            return position_symbol_pair.second.name() == edge_identifier.name_;
-                                          });
-    MG_ASSERT(position_on_frame != symbol_table.table().end());
     frame[symbol_table.at(edge_identifier)] = *e_acc;
```

Comment:
one more thing regarding to this: https://github.com/memgraph/memgraph/pull/578#discussion_r996815707

So you found the proper position/symbol for edge/node, but you are still using the "dummy" identifiers, so their position still will be wrong (-1). So I think edge_identifier / node_identifier should be gone, and edge_identifier_position_on_frame->second / node_identifier_position_on_frame->second should be used instead.

Reply:
Oh indeed that's completely correct. Well caught!